src/components/FilmList/FilmList.tsx

```diff
@@ -1,28 +1,58 @@
+import { useEffect, useState } from "react";
+import { filterByGenre, sorting } from "../../helpers";
 import { Film } from "../../interfaces";
 import { FilmCard } from "../FilmCard";
 
 import "./style.scss";
 
 interface FilmListProps {
+  selectedGenre: string;
+  sortBy: string;
+  setOpenedFilmId(filmID: string): void;
   movies: Film[];
   openModal(type: string, filmID: string): void;
 }
 
-const FilmList: React.FC<FilmListProps> = ({ movies, openModal }) => {
-  const filmsCards: any = movies.map((el) => {
-    return <FilmCard film={el} key={el.id} openModal={openModal} />;
-  });
+const FilmList: React.FC<FilmListProps> = ({
+  movies,
+  openModal,
+  selectedGenre,
+  sortBy,
+  setOpenedFilmId,
+}) => {
+  const [moviesConfigs, setMoviesConfigs] = useState(movies);
+
+  useEffect(() => {
+    setMoviesConfigs((prevState) => {
+      return [...sorting(prevState, sortBy)];
+    });
+  }, [sortBy]);
+
+  useEffect(() => {
+    setMoviesConfigs(filterByGenre(movies, selectedGenre));
+  }, [movies, selectedGenre]);
 
   const noFilm = <div>No movie found</div>;
   return (
     <>
       <div className="film-count">
         <h3>
-          <span>{movies.length}</span> movies found
+          <span>{moviesConfigs.length}</span> movies found
         </h3>
       </div>
       <div className={movies.length ? "film-list" : "film-list__none"}>
-        {movies.length ? filmsCards : noFilm}
+        {moviesConfigs.length
+          ? moviesConfigs.map((el) => {
+              return (
+                <FilmCard
+                  film={el}
+                  key={el.id}
+                  openModal={openModal}
```

Comment:
create better naming for props , we need to know what this openModal means (which modal.. etc)
and naming should start with 'on.....' , prefix and callback functions handlers should start with handle prefix
https://dlinau.wordpress.com/2016/02/22/how-to-name-props-for-react-components/
Check across whole app